### DISCUSSION TITLE
Getting lap times

### DISCUSSION THREAD

**Discussion by Oldertarl** [jolpica/jolpica-f1#112] [Q&A] [answered]
When I am trying to get the lap times for the different races, I am bumping in to some inconsistent data it looks like, or I am reading the data wrong - if so please advice 
when calling `https://api.jolpi.ca/ergast/f1/2024/1/laps/?format=json&limit=100&offset=0` I get the dataset for the first 100 lap times from the start of the race for every driver still active. Lap numbers start at 1, nothing wrong there 
than when getting the next dataset ( offset=1 ), lap numbers start again at 1, where to my understanding it should show lap number 6.

In addition, driver on position 1 (Max Verstappen) is not showing with offset=1 where in offset=0 the last driver is hulkenberg.

kindly advice if this is the data presentation or I am doing something wrong.

**Comment by jolpica** [comment, marked as answer]
Hey, 

Limit tells the API how kany results to return (in this case 100)
Offset tells the API at what result to start outputting from, not what multiple of the limit.

For example, if I have a list of numbers from 1 to 10:
limit=2,offset=0 would return 1,2
limit=2,offset=1 would return 2,3

So to get all the results in the smallest number of queries, you'd want to increase the offset in multiples of the limit
limit=2,offset=2 returns 3,4


This is different to the usual paging system for results that many other APIs use.

**Reply by Oldertarl** [reply]
How tired must I have been not getting that at start.
works indeed like a charm, so forget about me ever asking this please!
